### PR TITLE
WICKET-7011: Cleanup JUnit 5 assertions

### PR DESCRIPTION
**Pull request by vandeven**
Ran the Clean Up Assertions recipe from the OpenRewrite project (https://docs.openrewrite.org/reference/recipes/java/testing/junit5/cleanupassertions version 1.30.0).

Had to do 1 manual change to SpringBeanTest to make sure the == bean check was not replaced by assertEquals

### CODE REVIEW

**Review by martin-g — APPROVED** [review]

**Comment by timtebeek** [comment]
Glad to see this approved so quickly @martin-g ! :) Would you appreciate a few more such automated fixes?
I understand it's a fine balance to not overwhelm you with review requests when applying such automations.

Here's a few quick suggestions and their possible impact on the source code:

- [ ] [JUnit Jupiter best practices](https://docs.openrewrite.org/reference/recipes/java/testing/junit5/junit5bestpractices) - large changeset
- [ ] [Static imports for AssertJ](https://docs.openrewrite.org/reference/recipes/java/testing/assertj/staticimports) - small changeset
- [ ] [Migrate JUnit assertions to AssertJ](https://docs.openrewrite.org/reference/recipes/java/testing/assertj/junittoassertj) - large changeset
- [ ] [src/main code cleanup recipes](https://docs.openrewrite.org/reference/recipes/java/cleanup/cleanup) - large changeset
- [ ] [Logging best practices](https://docs.openrewrite.org/reference/recipes/java/logging) - large changeset
- [ ] [Adopt Java 17 code patterns](https://docs.openrewrite.org/reference/recipes/java/migrate) - small changeset

Typically the process is choosing a migration recipe, running a one-off command, visually inspecting the changes, running the tests and opening up a pull request.

Personally I find these automated migrations fun to do, especially on larger code bases, but like I said: I don't want to overwhelm for at times small improvements. Any feedback on what types of changes would (not) be appreciated would help to ensure a positive experience for all. :)

**Comment by martin-g** [comment]
> Would you appreciate a few more such automated fixes?

Sure!
Thank you!

File: wicket-spring/src/test/java/org/apache/wicket/spring/FieldBeansCollectorTest.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public void resolveGenericList() throws Exception
 
 		FieldBeansCollector fieldBeansCollector = new FieldBeansCollector(resolvableType);
 
-		assertTrue(fieldBeansCollector.getFieldType() == FieldType.LIST);
+		assertEquals(fieldBeansCollector.getFieldType(), FieldType.LIST);
```
Comment (solomax) [inline wicket-spring/src/test/java/org/apache/wicket/spring/FieldBeansCollectorTest.java:43]: arg positions also need to be be updated :)
are there any tools to check such errors? :)

`assertEquals(byte expected, byte actual, String message)`

Comment (timtebeek) [reply wicket-spring/src/test/java/org/apache/wicket/spring/FieldBeansCollectorTest.java:43]: I'd been meaning to write or look for a recipe on that; "if either argument is a literal, that argument should come first".

Comment (timtebeek) [reply wicket-spring/src/test/java/org/apache/wicket/spring/FieldBeansCollectorTest.java:43]: I'll let you know what I find & apply that here. :)

**Comment by martin-g** [comment]
Thank you, @vandeven & @timtebeek !

Please use https://issues.apache.org/jira/browse/WICKET-7011 as a ticket reference for any further work on this topic!

File: wicket-spring/src/test/java/org/apache/wicket/spring/FieldBeansCollectorTest.java (hunk 1 above)
Comment (timtebeek) [reply wicket-spring/src/test/java/org/apache/wicket/spring/FieldBeansCollectorTest.java:43]: Logged as a [possible improvement here](https://github.com/openrewrite/rewrite-testing-frameworks/issues/275). Once that's automated we can again apply it here, to also migrate any other imperfectly placed argument orders.

Comment (timtebeek) [reply wicket-spring/src/test/java/org/apache/wicket/spring/FieldBeansCollectorTest.java:43]: Picked up in https://github.com/apache/wicket/pull/546

**Pull request merged**